### DISCUSSION TITLE
Converting the cross section orientation to surface orientation in a sectioned grain

### DISCUSSION THREAD

**Discussion by Yasmine-G** [mtex-toolbox/mtex#1571] [Ask Anything] [answered]
Hi,

Is there any code available that can transform the EBSD orientation data taken from the cross section of a grain in a material to the orientation of that grain on the surface? Thanks a lot for your time!

**Comment by zmichels** [comment]
Do you mean that you want to rotate the orientations from your mapped-section by some angle (e.g., 90°) to result in a reference frame that matches a different surface?

If so... you can define the rotation  you want (i.e.,axis and angle) and apply it to a set of orientations.

Example:
% let "o" be the orientations you want to rotate
ax = vector3D.X
ang = 90*degree
rot = rotation('axis',ax,'angle',ang)
oRot = rot*o % the rotated result

apologize for any typos, I wrote this on my phone quick.

**Reply by zmichels** [reply]
Yeah looks like the asterisk symbol disappeared in the formatting of the post. 

90 * degree

rot * o

**Reply by Yasmine-G** [reply]
Thanks so much for your reply. Just one more question. How can I find the rotation axis?

**Comment by zmichels** [comment, marked as answer]
You will need to specify what axis in your map should be used. Perhaps it’s as simple as the X-axis (vector3D.X)… but it depends on your acquisition and MTEX-plotting preferences/settings. Maybe try plotting all the principal axes of the specimen reference frame separately and identify the one about which the rotation should be applied.

figure,
plot(vector3D.X)

figure,
plot(vector3D.Y)

figure,
plot(vector3D.Z)

**Reply by Yasmine-G** [reply]
Thanks a lot! I will try that.

**Reply by Yasmine-G** [reply]
Hi, 
Sorry to bother again. I am trying to run this code 
% for orientation as an example (111)
%% Rotation
o= Miller(1,1,1, cs);
ax = xvector
ang = 90 * degree;
rot = rotation('axis',ax,'angle',ang);
oRot = rot * o % the rotated result,
 
The answer I get is as shown h
![zz](https://user-images.githubusercontent.com/102839030/192054335-14eafb65-1d83-4b30-9cb3-6da20b0f08b6.JPG)
ere:

The first problem is the dot product is not 0. So they are not perpendicular!
The second problem is, at all cases, it only gives an answer which is a family plane!

Do you know what the problem is? Thanks for your time.

**Reply by zmichels** [reply]
Hi,

Previously you said you wanted to rotate orientations. However, in the code you shared, there are no orientations... instead you define o with a Miller direction. I'm not sure what you expected vs your result, but you won't get a rotated orientation from rotating a Miller axis. Maybe try with an orientation.

Here is a quick example of using it with the forsterite dataset:

%% load some example data
mtexdata forsterite

%% extract orientations and setup plotting axes
o = ebsd('f').orientations;
cs = o.CS;
h = [Miller(1,0,0,'direction',cs),Miller(0,1,0,'direction',cs),Miller(0,0,1,'direction',cs)];

%% plot initial un-rotated orientations in a pole figure
figure,
plotPDF(o,h','antipodal','lower','smooth','halfwidth',10*degree,'figSize','small')
![noRot](https://user-images.githubusercontent.com/8769619/192057013-774ed766-8178-456c-ac7a-f89735b489e9.png)

%% setup a rotation (I'll use same approach as in previous example)
ax = xvector;                           % axis
ang = 90 * degree;                      % angle
rot = rotation('axis',ax,'angle',ang);  % rotation

%% Apply the rotation
oRot = rot*o;

%% plot rotated orientation in a pole figure
figure,
plotPDF(oRot,h','antipodal','lower','smooth','halfwidth',10*degree,'figSize','small')
![Rot](https://user-images.githubusercontent.com/8769619/192057033-60d48c1f-ee7e-4a54-a32b-136a96151b9f.png)

**Reply by Yasmine-G** [reply]
Thanks very much for the example. It works perfectly fine and I can get the pole figures and rotate them to my surface orientation.
However, just to make sure this is the thing I should do, I cut a FIB sample which shows the cross section of my sample and I get  its EBSD data as shown in the image. 
![image](https://user-images.githubusercontent.com/102839030/192158794-d9190c2f-acbd-4d1f-9fb6-db0bd0a9f13d.png)
I want to know the orientation of every single grain on the surface separately based on the IPF color map. Is this the way to do that?
Thanks so much!

**Comment by lfgmorales** [comment]
Hi

What do you mean by „orientation of every single grain on the surface“? You mean the mean orientation of the grains from the bulk sample, and not only this fib cut?

Cheers

Luiz

Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 25 Sep 2022, at 20:34, Yasmine-G ***@***.***> wrote:

﻿

Thanks very much for the example. It works perfectly fine and I can get the pole figures and rotate them to my surface orientation.
However, just to make sure this is the thing I should do, I cut a FIB sample which shows the cross section of my sample and I get its EBSD data as shown in the image.
[image]<https://user-images.githubusercontent.com/102839030/192158794-d9190c2f-acbd-4d1f-9fb6-db0bd0a9f13d.png>
I want to know the orientation of every single grain on the surface separately based on the IPF color map. Is this the way to do that?
Thanks so much!

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1571#discussioncomment-3728739>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5INZ73NNWK4CYJRNWRTDWACLJVANCNFSM6AAAAAAQUBOO5U>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1571#discussioncomment-3728739", "url": "https://github.com/mtex-toolbox/mtex/discussions/1571#discussioncomment-3728739", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by Yasmine-G** [reply]
I mean only the FIB cut. I f I see a grain which is blue and has orientation close to (111), now what is the color and orientation on the surface. Thanks!

**Comment by hakonanes** [comment]
By "orientation", do you really mean crystal direction? As in if the grain is blue, it has one of its [uvw] = [111] close to parallel with the sample direction you chose when coloring your orientations (e.g. `inversePoleFigureDirection = zvector`)? If your attached middle image shows the cross section of your capping layer, I believe you want to plot the crystal directions parallel to the vertical in your map (perhaps `-yvector`, depending on your choice of Y sample direction, which you can check with `plot(yvector)`).
